In [ ]:
import pandas as pd
df = pd.read_csv('allbacks.csv')
df.shape

(15, 5)

In [ ]:
df.head(15)

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb
5,6,641,367,600,hb
6,7,1228,396,1075,hb
7,8,412,0,250,pb
8,9,953,0,700,pb
9,10,929,0,650,pb


In [56]:
df.sort_values(by=['weight'])
print('median weight:', df['weight'].median(), 'median cover:', 'pb')

median weight: 725.0 median cover: pb


The middle value of the data after being sorted (by weight) is 725.0 which belongs to a book with a paperback cover.

In [57]:
print('mean weight:', round(df['weight'].mean(), 2))

mean weight: 706.67


The average weight of all the books is 706.67 grams

In [58]:
print('weights_by_cover:', round(df.groupby('cover')['weight'].mean(), 2))

weights_by_cover: cover
hb    796.43
pb    628.12
Name: weight, dtype: float64


The average weight of hardcover books are 796.43 grams and the average weight of paperback books are 638.12 grams.

**manual calculation for mean of hardcoverbooks:**

mean: [800 + 950 + 1050 + 350 + 750 + 600 + 1075]/y = 5575/7 = 796.43